api/client.go

```diff
@@ -622,11 +622,19 @@ func (c *Client) Headers() http.Header {
 	return ret
 }
 
-// SetHeaders sets the headers to be used for future requests.
+// AddHeader allows a single header key/value pair to be added
+// in a race-safe fashion.
+func (c *Client) AddHeader(key, value string) {
+	c.modifyLock.RLock()
```

Comment:
Should this be `c.modifyLock.Lock()`
I'm probably way off on RWMutex sync patterns, but I thought that the writer used a different method.

Reply:
YES it should. Great catch! Thank you!

Reply:
Fixed in https://github.com/hashicorp/vault/pull/7999/commits/7a7b816124b21243777a0c97cb309ef3068312a4

Reply:
👍